src/lib/auth/user.ts

```diff
@@ -3,17 +3,26 @@ import { cache } from "react"
 import { cmp, db, schema } from "../db"
 import { logger } from "../server/logger"
 import { createServerClient } from "../supabase/server"
+import { type User } from "@supabase/supabase-js"
 
-async function getAuthUserImpl() {
+export const ADMIN_EMAILS = ["soorria.ss@gmail.com", "ericcpaul00@gmail.com"]
+
+async function getAuthUserImpl(): Promise<User | null> {
   const sb = createServerClient()
   const { data, error } = await sb.auth.getUser()
 
-  if (error) {
+  if (data.user === null || error) {
+    logger.error(error, "failed to get user")
+    return null
+  }
+
+  const dbUser = await getDBUser(data.user.id);
```

Comment:
may not want to do this since how it makes every usage of `getAuthUserImpl` require a db query

Reply:
if we need to check if a user is admin, we can make it a separate function

Reply:
good shout